### PR TITLE
Mapflags

### LINKED ISSUE #318: Object flags in MAP
We should read and set object flags, light values, etc. from MAP and set it for object.
They override values from prototype.

### PR DESCRIPTION
**Pull request by isage**
1. read object flags from map and override flags from proto
2. read light values from map and override values from proto
3. read/set "flat" flag and render flat objects first.

Closes #318

### CODE REVIEW

File: src/Game/Object.cpp [hunk 1]
```diff
@@ -529,5 +529,52 @@ unsigned short GameObject::lightOrientation() const
...
+{
+    _lightRadius = radius;
+}
+
+unsigned int GameObject::lightRadius() const
+{
+    return _lightRadius;
+}
+
+void GameObject::parseFlags(unsigned int flags)
```
Comment (phobos2077) [inline src/Game/Object.cpp]: Maybe it's better to name it *setFlags* and change type to uint32_t ?

File: src/Game/ObjectFactory.cpp [hunk 2]
```diff
@@ -308,18 +308,7 @@ GameObject* GameObjectFactory::createObject(unsigned int PID)
     }
     object->setPID(PID);
     object->setFID(proto->FID());
-    object->setCanWalkThru(proto->flags() & 0x00000010);
-    object->setCanLightThru(proto->flags() & 0x20000000);
-    object->setCanShootThru(proto->flags() & 0x80000000);
-
-    object->setTrans(GameObject::TRANS_DEFAULT);
-    if (proto->flags() & 0x00004000) object->setTrans(GameObject::TRANS_RED);
```
Comment (alexeevdv) [inline src/Game/ObjectFactory.cpp]: So.... there are no more flags from PRO file? I thought that flags should be overrided by MAP file, not replaced :) Some objects are created directly from scripts using PRO files

File: src/Game/Object.cpp (hunk 1 above)
Comment (alexeevdv) [inline src/Game/Object.cpp]: +1 for *setFlags*
We don't need uint32_t here, cause we performing using binary AND operation on flags value

File: src/Game/ObjectFactory.cpp (hunk 2 above)
Comment (phobos2077) [inline src/Game/ObjectFactory.cpp]: He called parseFlags() below, which does the same as the replaced code, as far as I can see.

File: src/Game/Object.h [hunk 3]
```diff
@@ -182,6 +185,17 @@ class GameObject : public EventEmitter
...
+
+    unsigned int lightIntensity() const;
+    virtual void setLightIntensity(unsigned int intensity);
+
+    unsigned int lightRadius() const;
+    virtual void setLightRadius(unsigned int radius);
+
+    virtual void parseFlags(unsigned int flags);
+
+    bool flat();
```
Comment (alexeevdv) [inline src/Game/Object.h]: const :)

File: src/Game/ObjectFactory.cpp (hunk 2 above)
Comment (alexeevdv) [inline src/Game/ObjectFactory.cpp]: Yep. Missed that

File: src/Game/Object.cpp [hunk 4]
```diff
@@ -529,5 +529,52 @@ unsigned short GameObject::lightOrientation() const
...
+{
+    _lightRadius = radius;
+}
+
+unsigned int GameObject::lightRadius() const
+{
+    return _lightRadius;
+}
+
+void GameObject::setFlags(unsigned int flags)
```
Comment (phobos2077) [inline src/Game/Object.cpp]: how about fixed-width data type for flags?

Comment (alexeevdv) [inline src/Game/Object.cpp]: There is no need for this as far as i can see

**Pull request merged**